### PR TITLE
Implement difference between urban and rural for events and roads

### LINKED ISSUE #207: Change Colour coding for overtaking distance and show statistic distribution of events per street segment
All single events, average or median overtaking distances in street segments below 150 cm inside and 200 cm outside off towns should be marked at least in light red or orange. Only distances clearly above 150 cm in town (maybe above 175 cm out of towns) should be marked in yellow and green. 
My suggestion (in / out of town):
- below 100 / 125 cm - dark red
- 100 - 125 / 125 - 150 cm - red
- 125 - 150 / 150 - 175 cm light red / orange
- 150 - 175 / 175 - 200 cm yellow
- 175 - 200 / 200 - 225 cm light green
- above 200 / 225 cm green
I am aware that the sensors will not show overtaking distances far above 250 cm, so this is why the colour scale IMO could also be a bit less hard taking into account there could be some events that have not been measured. But no green below the STVO-Limit.
In addition I would find it very helpful to get back the bar graph that shows the distribution of overtaking events in a street segment.

### PR DESCRIPTION
**Pull request by gluap**
@opatut This is my take at enabling the differentiation between "urban" and "rural" for the portal-based map.

It will require:
- schema upgrade (alembic) to add the "road" column to the overtaking_event table
- prepare_sql_tiles to change the tile schema (took me a while that dis determines what data is present in the tile)
- re-computation the events

Left for the future:
- Switching to the ruralness AI for categorizing non-tagged roads

Looks like this (note: border color on events conveys ``urbanity``, histogram redness threshold shifts to two meters for rural.:
image
image

fixes #207
improves previous fixes to #118

### CODE REVIEW

**Comment by opatut** [comment]
Why copy the zone to the event and not JOIN it in from the road? That way we could later improve zone detection for roads without reimport, and we'd have less duplicated data too.

**Comment by opatut** [comment]
I don't like the border color at all, it distorts the visuals of the actual data IMO. I don't think this should be shown, maybe just have a "detected zone" option in the dropdown of the road coloring. After all, that isn't really needed for evaluation, mostly just for debugging, I think.

**Comment by gluap** [comment]
> I don't like the border color at all
In the current state only a fraction of the rural roads are tagged. As long as this is the case not visualizing that fact in some way gives a false sense of "overtaking distance is kept by most drivers". It could be made toggleable.

File: tile-generator/layers/obs_events/layer.sql [hunk 1]
```diff
@@ -9,8 +9,10 @@ RETURNS TABLE(event_id bigint, geometry geometry, distance_overtaker float, dist
       (case when direction_reversed then -1 else 1 end)::int as direction,
       course,
       speed,
-      way_id::bigint as way_id
+      CASE WHEN road.zone IS NULL THEN 'urban' else road.zone END as zone,
+      overtaking_event.way_id::bigint as way_id
     FROM overtaking_event
+    FULL OUTER JOIN road ON (road.way_id = overtaking_event.way_id)
```
Comment (gluap) [inline tile-generator/layers/obs_events/layer.sql:15]: taking the zone from road here as suggested by opatut in the review.

File: tile-generator/layers/obs_events/layer.sql [hunk 2]
```diff
@@ -9,8 +9,10 @@ RETURNS TABLE(event_id bigint, geometry geometry, distance_overtaker float, dist
...
     WHERE ST_Transform(overtaking_event.geometry, 3857) && bbox;
```
Comment (gluap) [inline tile-generator/layers/obs_events/layer.sql:17]: @opatut this way it works without the duplicate column - thank you for the suggestion, that made things way nicer.

**Comment by gluap** [comment]
Screenshots of the latest iteration: 
Mode to show assumed lawful overtaking distance:
image

**Comment by opatut** [comment]
@gluap, can I get this "latest iteration" please ;)

**Comment by gluap** [comment]
@opatut the "latest iteration" should now be in. Thank you for the git magic allowing me to recover it.

**Comment by gluap** [comment]
@opatut the latest rural/urban view can now also be tested on https://obs.adfc-darmstadt.de

**Review by opatut — APPROVED** [review]

**Pull request merged**